pkg/cri/containerd/runtime.go

```diff
@@ -0,0 +1,72 @@
+package containerd
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/containerd/oci"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+const k8sNamespace = "k8s.io"
+
+// ContainerdRuntime represents a connection to the containerd runtime
+type ContainerdRuntime struct {
+	client            Client
+	namespacedContext context.Context
+}
+
+// NewContainerdRuntime connects to the containerd runtime over the specified `socketPath`
+func NewContainerdRuntime(socketPath string, timeout time.Duration) (*ContainerdRuntime, error) {
+	containerdRuntime, err := containerd.New(
+		socketPath,
+		containerd.WithTimeout(timeout),
+		containerd.WithDefaultNamespace(k8sNamespace))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create containerd client: %w", err)
+	}
+
+	return newContainerdRuntime(containerdRuntime), nil
+}
+
+func newContainerdRuntime(client Client) *ContainerdRuntime {
+	return &ContainerdRuntime{
+		client:            client,
+		namespacedContext: namespaces.WithNamespace(context.Background(), k8sNamespace),
+	}
+}
+
+// NetNS returns the netns path of a given container
+func (cd *ContainerdRuntime) NetNS(containerID string) (string, error) {
```

Comment:
```golang
func (cr *ContainerdRuntime) FindNetNsByContainerID(containerID string) (string, error) {
```

Reply:
Since the type is a `string` you have to state what this string is, if you do `NetNS("foo")` you don't know what that foo has to be when reading the code.